sympy/matrices/dense.py

```diff
@@ -35,15 +36,16 @@ def _compare_sequence(a, b):
     # tuple
     return tuple(a) == tuple(b)
 
+@sympify_method_args
 class DenseMatrix(MatrixBase):
 
     is_MatrixExpr = False  # type: bool
 
     _op_priority = 10.01
     _class_priority = 4
 
+    @sympify_return([('other', 'DenseMatrix')], NotImplemented)
     def __eq__(self, other):
-        other = sympify(other)
         self_shape = getattr(self, 'shape', None)
         other_shape = getattr(other, 'shape', None)
         if None in (self_shape, other_shape):
```

Comment:
The conditions below suggest that sometimes `other` is not a `DenseMatrix`. Is that not ever the case?

Reply:
It works for all cases .